components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -180,23 +180,57 @@ def _process_s3_input(
         paths_to_compress_buffer.add_file(object_metadata)
 
 
+def _write_failed_compression_log(
+    status_msg: str, logs_directory: Path, job_id: Any
+) -> Optional[Path]:
+    """
+    Writes status_msg to a new log file located at
+    `{logs_directory}/user/failed_compression_log_{job_id}.txt`. The /user directory will be created
+    if it doesn't already exist.
+    :param status_msg:
+    :param logs_directory:
+    :param job_id:
+    :return: Path to the written log file relative to {logs_directory}, or `None` on error.
+    """
+
+    user_logs_dir = Path(logs_directory) / "user"
+    try:
+        user_logs_dir.mkdir(parents=True, exist_ok=True)
+    except Exception as e:
+        logger.warning(f"Failed to create user logs directory: {e}")
+        return None
+
+    log_path = user_logs_dir / f"failed_compression_log_{job_id}.txt"
+    try:
+        with log_path.open("w", encoding="utf-8") as f:
+            timestamp = datetime.datetime.now().isoformat(timespec="seconds")
+            f.write(f"Failed compression job log.\nGenerated at {timestamp}.\n\n")
+            f.write(f"{status_msg.rstrip()}\n")
+    except Exception as e:
+        logger.warning(f"Failed to write compression failure log: {e}")
+        return None
+
+    return Path("user") / f"failed_compression_log_{job_id}.txt"
+
+
 def _write_failed_path_log(
     invalid_path_messages: List[str], logs_directory: Path, job_id: Any
 ) -> Optional[Path]:
     """
-    Writes the error messages in `invalid_path_messages` to a log file,
-    `{logs_directory}/user/failed_paths_{job_id}.txt`. The directory will be created if it doesn't
-    already exist.
+    Writes the error messages in `invalid_path_messages` to a new log file located at
+    `{logs_directory}/user/failed_paths_{job_id}.txt`. The /user directory will be created if it
+    doesn't already exist.
     :param invalid_path_messages:
     :param logs_directory:
     :param job_id:
-    :return: Path to the written log file or `None` if error is encountered.
+    :return: Path to the written log file relative to {logs_directory}, or `None` on error.
     """
 
     user_logs_dir = Path(logs_directory) / "user"
     try:
         user_logs_dir.mkdir(parents=True, exist_ok=True)
-    except Exception:
+    except Exception as e:
+        logger.warning(f"Failed to create user logs directory: {e}")
```

Comment:
```suggestion
        logger.error(f"Failed to create directory: '%s' - %s", user_logs_dir, e)
```

Should be an error, right?